src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -496,13 +499,15 @@ public void analyse(Network network, List<PropagatedContingency> contingencies,
      * The dimensions of both matrices must be the same
      */
     // TODO : implement this method for DenseMatrix in powsybl-core ?
-    private static void matrixShallowCopy(DenseMatrix originalMatrix, DenseMatrix copyMatrix) {
+    private static void matrixCopyValues(DenseMatrix originalMatrix, DenseMatrix copyMatrix) {
         if (originalMatrix.getRowCount() == copyMatrix.getRowCount() && originalMatrix.getColumnCount() == copyMatrix.getColumnCount()) {
             for (int columnIndex = 0; columnIndex < originalMatrix.getColumnCount(); columnIndex++) {
                 for (int rowIndex = 0; rowIndex < originalMatrix.getRowCount(); rowIndex++) {
                     copyMatrix.set(rowIndex, columnIndex, originalMatrix.get(rowIndex, columnIndex));
                 }
             }
+        } else {
+            throw new IllegalArgumentException();
         }
```

Comment:
Maybe a message in the exception. For example. Invalid matrix size for copyMatrix. x columns and r rows expected. Received x' columns and r' rows.

Reply:
Yes, I modified this exception to a MatrixException (same check is done and same exception is used when summing two matrices in powybl-core)